python/mdvtools/llm/chat_server_extension.py

```
@@ -62,38 +68,59 @@ def chat_init():
-            if not request.json:
-                return {"error": "No JSON data in request"}, 500
-            message = request.json.get("message")
-            id = request.json.get("id")
+            sid = request.sid # type: ignore - flask_socketio.request.sid
+            # todo - auth (via custom decorator? see comments above)
+
+            message = data.get("message")
+            id = data.get("id")
+            room = f"{sid}_{id}"
+            join_room(room)
+            def handle_error(error: str):
```

Flake8 (7.2.0)

[error] expected 1 blank line before a nested definition, found 0

(E306)